### PR TITLE
Feature 65

### LINKED ISSUE #65: As a guardian I would like to be able to customize how a completed activity is shown in the week plan for a given citizen, so that it fits their personal preference
I would like:

- [ ] A check mark
- [ ] Removal of the activity
- [ ] Making the activity translucent
- [ ] Making the activity greyed out

### PR DESCRIPTION
**Pull request by elaurs**
fixes #65 

We have removed the lines from the covered completed icons. There is still an issue getting the device to not show the lines on iOS devices when displaying 5 days in the weekplanner. Else then that all the lines should be gone. We have created an issue for this.

### CODE REVIEW

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw613f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [ ] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [ ] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [ ] The features that the code claims to implement, are actually implemented
- [ ] The code does not look like it has bugs
- [ ] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by mads-bo — COMMENTED** [review]
@aau-giraf/sw614f20 you have been assigned this. Here's your checklist.
Please use the following checklist. Feel free to check it off as you go along.

**Code Design**

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems
- [x] The code does it introduce functionality that is not necessary for solving the problem

**Code Readability and Maintainability**

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no "hack", or obscure solution
- [ ] All new functionality is covered by tests
- [x] It is understandable, by reading the tests, what they do
- [ ] The tests covers sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing
- [ ] Error messages in the code are understandable
- [ ] Confusing sections of code are covered by comments

**Functionality**

For this part, open the app, and test:

- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Points for Discussion and Reflection**

- [ ] UI messages, that the user might meet, are actually helpful
- [ ] There are no errors that are bad practice. E.g. hard-coding a test database connection
- [ ] There is well written code in this. Remember to tell the developer!

**Review by Lortemanden — CHANGES_REQUESTED** [review]
The setting where the activity is removed doesn't work as intended. It should not be possible for the citizens to click on the activity and enter it after it's been completed. Also the colour that has been put on top of the activity don´t cover it all. 

The greyed out setting works fine. We just think it should be more clearly greyed out.

File: lib/screens/settings_screens/completed_activity_icon_selection_screen.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import 'package:flutter/material.dart';
+import 'package:weekplanner/blocs/settings_bloc.dart';
+import 'package:weekplanner/widgets/giraf_app_bar_widget.dart';
+import 'package:weekplanner/widgets/settings_widgets/settings_section.dart';
+import 'package:weekplanner/widgets/settings_widgets/settings_section_checkboxButton.dart';
+import 'package:weekplanner/widgets/settings_widgets/settings_section_item.dart';
+
+import '../../di.dart';
+
+/// Screen where the user can select how many days to show for a citizen
```
Comment (Simon97) [inline lib/screens/settings_screens/completed_activity_icon_selection_screen.dart]:
```suggestion
/// Screen where the icon for completed activity can be chosen
```

File: lib/screens/settings_screens/completed_activity_icon_selection_screen.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                AsyncSnapshot<SettingsModel> settingsSnapshot) {
+              if (settingsSnapshot.hasData) {
+                final SettingsModel _settingsModel = settingsSnapshot.data;
+
+                return ListView(
+                  children: <Widget>[
+                    SettingsSection('Tegn for udførelse', <SettingsSectionItem>[
+                      SettingsCheckMarkButton(
+                          1,
+                          _settingsModel.completeMark.index + 1,
```
Comment (Simon97) [inline lib/screens/settings_screens/completed_activity_icon_selection_screen.dart]: Hvorfor er det at der plusses med 1?
Kunne man ikke bare sammenligne den anden parameter til SettingsCheckMarkButton med 0 i stedet? :) Altså så expected i den første checkMark-knap er 0 istedet for 1?

Det samme spørgsmål er relevant for de andre CheckMarkKnapper

**Review by Simon97 — CHANGES_REQUESTED** [review]
Når man vælger at fjerne aktiviteterne, som markering for at de er færdige, kan man stadig se ikonet for at aktiviteten har en timer. Det tænker jeg umiddelbart ikke at man skal kunne? :)

**Comment by elaurs** [comment]
Vi har ikke kunne fjerne stregerne helt fra de fjernede completed ikoner. Det burde dog kun være iOS på den setting med at vise 5 dage, hvor stregerne ikke er væk. Det ser ud til at virke på alle de andre. Vi har lavet et nyt issue til det.

**Review by Simon97 — APPROVED** [review]
Antaget at PO og kunderne synes det er fint er det ligemeget, men jeg synes det virker underligt at citizens skal scrolle i nogle aktiviteter der i princippet ikke er på listen længere. Der kan ende med at være lidt langt ned til de nederste.
Men det må være op til PO, så jeg approver. :)

**Review by Lortemanden — APPROVED** [review]

**Pull request merged**